salt/modules/mount.py

```diff
@@ -1422,3 +1569,287 @@ def delete_mount_cache(real_name):
+    to match the data passed, or add the mount if it is not present on AIX
+
+        Provide information if the path is mounted
+
+    :param name:          The name of the mount point where the device is mounted.
+    :param device:        The device that is being mounted.
+    :param vfstype:       The file system that is used (AIX has two fstypes, fstype and vfstype - similar to Linux fstype)
+    :param opts:          Additional options used when mounting the device.
+    :param mount:         Mount if not mounted, default True.
+    :param config:        Configuration file, default /etc/filesystems.
+    :param match:         File systems type to match on, default auto
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mount.set_filesystems /mnt/foo /dev/sdz1 jfs2
+    '''
+    # Fix the opts type if it is a list
+    if isinstance(opts, list):
+        opts = ','.join(opts)
+
+    # preserve arguments for updating
+    entry_args = {
+        'name': name,
+        'dev': device.replace('\\ ', '\\040'),
+        'vfstype': vfstype,
+        'opts': opts,
+        'mount': mount,
+    }
+
+    view_lines = []
+    ret = None
+
+    if 'AIX' not in __grains__['kernel']:
+        return ret
+
+    # Transform match_on into list--items will be checked later
+    if isinstance(match_on, list):
+        pass
+    elif not isinstance(match_on, six.string_types):
+        msg = 'match_on must be a string or list of strings'
+        raise CommandExecutionError(msg)
+    elif match_on == 'auto':
+        # Try to guess right criteria for auto....
+        # added IBM types from sys/vmount.h after btrfs
+        # NOTE: missing some special fstypes here
+        specialFSes = frozenset([
+            'none',
+            'tmpfs',
+            'sysfs',
+            'proc',
+            'fusectl',
+            'debugfs',
+            'securityfs',
+            'devtmpfs',
+            'cgroup',
+            'btrfs',
+            'cdrfs',
+            'procfs',
+            'jfs',
+            'jfs2',
+            'nfs',
+            'sfs',
+            'nfs3',
+            'cachefs',
+            'udfs',
+            'cifs',
+            'namefs',
+            'pmemfs',
+            'ahafs',
+            'nfs4',
+            'autofs',
+            'stnfs'])
+
+        if vfstype in specialFSes:
+            match_on = ['name']
+        else:
+            match_on = ['dev']
+    else:
+        match_on = [match_on]
+
+    # generate entry and criteria objects, handle invalid keys in match_on
+    entry_ip = _filesystems_entry.from_line(entry_args, kwargs)
+    try:
+        criteria = entry_ip.pick(match_on)
+
+    except KeyError:
+        filterFn = lambda key: key not in _filesystems_entry.compatibility_keys
+        invalid_keys = filter(filterFn, match_on)
+
+        msg = 'Unrecognized keys in match_on: "{0}"'.format(invalid_keys)
+        raise CommandExecutionError(msg)
```

Comment:
We don't need to assign this to a variable if all we're doing is raising an exception with it.

Reply:
@terminalmage but automaster was written as a complement to fstab for Darwin.  As for msg, adjusted but similar usage throughout file